java/daikon/dcomp/CalcStack24.java

```diff
@@ -0,0 +1,1047 @@
+          case Opcode.NEWARRAY:
+            stack.pop(); // discard the count
+            final NewPrimitiveArrayInstruction npai = (NewPrimitiveArrayInstruction) inst;
+            String descriptor;
+            switch (npai.typeKind()) {
+              case BOOLEAN -> {
+                descriptor = "[Z";
+              }
+              case BYTE -> {
+                descriptor = "[B";
+              }
+              case CHAR -> {
+                descriptor = "[C";
+              }
+              case DOUBLE -> {
+                descriptor = "[D";
+              }
+              case FLOAT -> {
+                descriptor = "[F";
+              }
+              case INT -> {
+                descriptor = "[I";
+              }
+              case LONG -> {
+                descriptor = "[J";
+              }
+              case SHORT -> {
+                descriptor = "[S";
+              }
+              default -> {
+                throw new DynCompError("unknown primitive array type: " + inst);
+              }
+            }
+            stack.push(ClassDesc.ofDescriptor(descriptor));
+            return true;
+
+          // operand stack:
+          // ..., value
+          // ...
+          case Opcode.POP:
+            stack.pop(); // discard the value
+            return true;
+
+          // operand stack:
+          // ..., value2, value1
+          // ...
+          // where each of value1 and value2 is a category 1 computational type
+          // OR
+          // ..., value
+          // ...
+          // where value is of a category 2 computational type
+          case Opcode.POP2:
+            {
+              final ClassDesc v1 = stack.pop();
+              if (stack.slotSize(v1) == 1) {
+                stack.pop();
+              }
+              return true;
+            }
+
+          // operand stack:
+          // ..., objectref, value
+          // ...
+          case Opcode.PUTFIELD:
+            stack.pop(2);
+            return true;
+
+          // operand stack:
+          // ..., value
+          // ...
+          case Opcode.PUTSTATIC:
+            stack.pop(); // discard the value
+            return true;
+
+          // TODO: JSR and RET have been illegal since JDK 7 (class file version 51.0).
+          // Consider throwing an UnsupportedOperationException for these opcodes
+          // if we're only supporting modern class files (version 51.0+).
+          // Currently maintaining support for completeness.
+
+          // operand stack:
+          // [no change]
+          case Opcode.RET:
+          case Opcode.RET_W:
+            // the variable referenced must contain a return address
+            // that we would treat as target of jump; no way to do that
+            return false;
+
+          // operand stack:
+          // ..., value1, value2
+          // ..., value2, value1
+          case Opcode.SWAP:
+            {
+              final ClassDesc v2 = stack.pop();
```

Comment:
Minor: `v1` and `v2` are actually types (well, ClassDesc) rather than values.  Thus, the use of variable names `v` is a bit misleading.  But, be consistent throughout the whole file.

Reply:
All of the 'operand stack before:' and 'operand stack after:' comments were taken straight from the The Java Virtual Machine Instruction Set chapter of the Java Virtual Machine Specification.  They use value1, value2, etc.  So I think using the short hand v1, v2, etc. makes sense.  Are there places where I don't follow that convention?